metadata/metadata.go

```diff
@@ -127,17 +127,17 @@ func FromContext(ctx context.Context) (md MD, ok bool) {
 	return FromIncomingContext(ctx)
 }
 
-// FromIncomingContext returns the incoming MD in ctx if it exists.  The
-// returned md should be immutable, writing to it may cause races.
-// Modification should be made to the copies of the returned md.
+// FromIncomingContext returns the incoming metadata in ctx if it exists.  The
+// returned MD should be immutable. Writing to it may cause races.
```

Comment:
should be immutable -> should not be modified.

It is NOT immutable, and this might lead someone to believe that what is returned is a copy of the data and that modifying it is fine but doesn't change the underlying metadata (neither of which is true).

(Also in FromOutgoingContext.)